### PR TITLE
[FLINK-21785][table-planner-blink] Support json ser/de for StreamExecCorrelate

### PR DESCRIPTION
**Pull request by wenlong88**
## What is the purpose of the change

Support json ser/de for StreamExecCorrelate

## Brief change log
Support json ser/de for StreamExecCorrelate

## Verifying this change

This change added tests and can be verified as follows:

  - *Added test that validates json ser/de in CorrelateJsonPlanTest and CorrelatePlanJsonITCase*
  
## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecCorrelate.java [hunk 1]
```diff
@@ -53,4 +60,25 @@ public StreamExecCorrelate(
                 outputType,
                 description);
```
Comment (zjuwangg) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecCorrelate.java:61]: nit： use this()

**Review by zjuwangg — APPROVED** [review]

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecCorrelate.java [hunk 2]
```diff
@@ -58,7 +76,29 @@ public CommonExecCorrelate(
...
+    }
+
+    public CommonExecCorrelate(
+            FlinkJoinType joinType,
+            RexCall invocation,
+            @Nullable RexNode condition,
+            Class<?> operatorBaseClass,
+            boolean retainHeader,
+            int id,
+            List<InputProperty> inputProperty,
```
Comment (godfreyhe) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecCorrelate.java:98]: inputProperties

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala [hunk 3]
```diff
@@ -746,11 +746,12 @@ abstract class TableTestUtilBase(test: TableTestBase, isStreamingMode: Boolean)
     val file = new File(s"./src/test/resources/$path/$fileName")
     if (file.exists()) {
       val expected = TableTestUtil.readFromResource(s"$path/$fileName")
+      val mapper = new ObjectMapper();
       assertEquals(
-        TableTestUtil.replaceExecNodeId(
-          TableTestUtil.getFormattedJson(expected)),
-        TableTestUtil.replaceExecNodeId(
-          TableTestUtil.getFormattedJson(jsonPlanWithoutFlinkVersion)))
+        mapper.readTree(TableTestUtil.replaceExecNodeId(
```
Comment (godfreyhe) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala:751]: we do not format the json if we use json node to compare.

**Review by godfreyhe — COMMENTED** [review]
Thanks for the contribution @wenlong88 , LGTM overall, I left two minor comments

**Review by godfreyhe — APPROVED** [review]

**Pull request closed without merging**